addons/web/static/src/core/debug/profiling/profiling_qweb.js

```diff
@@ -23,9 +23,12 @@ export class ProfilingQwebView extends Component {
         this.ace = useRef("ace");
         this.selector = useRef("selector");
 
-        for (const line of this.profile.data) {
+        let data = JSON.parse(this.props.value);
+        for (const line of data[0].results.data) {
             line.xpath = line.xpath.replace(/([^\]])\//g, "$1[1]/").replace(/([^\]])$/g, "$1[1]");
         }
+        this.props.value = JSON.stringify(data);
```

Comment:
Writing on props is an owl anti pattern and must be avoided. The owner of the props is the parent, not the component itself, this could lead to confusion. I suggest to get rid of the getter instead, and to do something like
```js
this.profile = this.computeProfile(this.props.value);
onWillUpdateProps(() => {
    this.profile = this.computeProfile(nextProps.value);
});
```
with a `computeProfile` function doing the parsing/replacing stuff.

Reply:
You kindly suggested to do it better yourself.
Is it always in your plan ?

Reply:
We can take it over indeed.

Reply:
@FrancoisGe